### DISCUSSION TITLE
Inquiry regarding reproduction of Table 3 and model saving.

### DISCUSSION THREAD

**Discussion by lim-sorry** [bjzhb666/GS-LoRA#14] [Q&A] [answered]
Hello, thank you for sharing your great research and code. I am interested in your work and currently trying to reproduce the pre-trained results in Table 3 using the Face100 dataset and the pre-trained model (dated 2024-09-26) mentioned in your paper. However, I have encountered some discrepancies.

1. Pre-trained Result Discrepancy: When I evaluated the pre-trained model on the test set without any further training, the results for $Acc_r$ and $Acc_f$ were consistent with the "Class Missing Forgetting" results shown in Table 5, rather than Table 3. Could you please clarify how the pre-trained results in Table 3 were calculated? I would appreciate it if you could correct any misunderstandings I might have. 
2. Model Selection for Best Results:Were the "Best" results calculated based on the model that achieved the highest Harmonic Mean (H-mean) for each task? In the current code, it appears that when transitioning from Task 0 to Task 1, the model from the last epoch—rather than the one with the best H-mean—is saved as backbone_task_n.pth. I would like to know how you handled model selection during the incremental learning process.

Again, thanks for your sharing.

```
...
test_dataset = datasets.ImageFolder(
    root="./data/faces_webface_112x112_sub100_train_test/test",
    transform=data_transform,
)
...
order_list = list(range(100))
random.seed(1337)
random.shuffle(order_list)

for order in range(80,0,-20):
    retain = order_list[:order]
    forget = order_list[order:order+20]

    retain_correct = 0
    retain_total = 0    
    for i in retain:
        retain_correct += class_correct[i]
        retain_total += class_total[i]
    print("Retain Accuracy: %4.4f %%\n" % (100 * retain_correct / retain_total))
    
    forget_correct = 0
    forget_total = 0
    for i in forget:
        forget_correct += class_correct[i]
        forget_total += class_total[i]
    print("Forget Accuracy: %4.4f %%\n" % (100 * forget_correct / forget_total))

""" results
Retain classes: [83, 17, 10, 9, 89, 52, 7, 32, 37, 77, 61, 34, 78, 55, 56, 63, 69, 22, 75, 66, 35, 72, 71, 23, 86, 38, 24, 5, 29, 30, 16, 11, 82, 19, 57, 14, 91, 27, 98, 53, 0, 20, 59, 28, 47, 18, 6, 48, 12, 62, 1, 43, 3, 60, 41, 85, 4, 95, 97, 36, 94, 65, 67, 99, 25, 33, 70, 40, 92, 31, 58, 15, 45, 2, 87, 76, 80, 44, 64, 8]
Forget classes: [51, 54, 13, 88, 84, 26, 50, 39, 96, 81, 49, 42, 21, 93, 74, 73, 46, 90, 68, 79]
Retain Accuracy: 73.5048 %
Forget Accuracy: 74.4473 %

Retain classes: [83, 17, 10, 9, 89, 52, 7, 32, 37, 77, 61, 34, 78, 55, 56, 63, 69, 22, 75, 66, 35, 72, 71, 23, 86, 38, 24, 5, 29, 30, 16, 11, 82, 19, 57, 14, 91, 27, 98, 53, 0, 20, 59, 28, 47, 18, 6, 48, 12, 62, 1, 43, 3, 60, 41, 85, 4, 95, 97, 36]
Forget classes: [94, 65, 67, 99, 25, 33, 70, 40, 92, 31, 58, 15, 45, 2, 87, 76, 80, 44, 64, 8]
Retain Accuracy: 73.2192 %
Forget Accuracy: 74.2767 %

Retain classes: [83, 17, 10, 9, 89, 52, 7, 32, 37, 77, 61, 34, 78, 55, 56, 63, 69, 22, 75, 66, 35, 72, 71, 23, 86, 38, 24, 5, 29, 30, 16, 11, 82, 19, 57, 14, 91, 27, 98, 53]
Forget classes: [0, 20, 59, 28, 47, 18, 6, 48, 12, 62, 1, 43, 3, 60, 41, 85, 4, 95, 97, 36]
Retain Accuracy: 72.9408 %
Forget Accuracy: 73.7870 %

Retain classes: [83, 17, 10, 9, 89, 52, 7, 32, 37, 77, 61, 34, 78, 55, 56, 63, 69, 22, 75, 66]
Forget classes: [35, 72, 71, 23, 86, 38, 24, 5, 29, 30, 16, 11, 82, 19, 57, 14, 91, 27, 98, 53]
Retain Accuracy: 73.3412 %
Forget Accuracy: 72.5568 % """
```

**Comment by bjzhb666** [comment, marked as answer]
Thanks for your interest! We recommand you to use the results in the latest (journal version). We will update the final journal version in two days.

We use the last epoch not the best, you are right.
